api/test/integration/indicator-record/indicator-records.service.spec.ts

```diff
@@ -0,0 +1,364 @@
+
+      const sourcingData = {
+        sourcingRecordId: indicatorPreconditions.sourcingRecord2.id,
+        tonnage: indicatorPreconditions.sourcingRecord2.tonnage,
+        geoRegionId: indicatorPreconditions.sourcingLocation2.geoRegionId,
+        materialId: indicatorPreconditions.sourcingLocation2.materialId,
+      };
+
+      // The original function only expects for one MaterialTo3 data, the values don't matter
+      jest.spyOn(materialsToH3sService, 'findOne').mockResolvedValue({
+        materialId: UUIDv4(),
+        id: UUIDv4(),
+        h3DataId: UUIDv4(),
+        ...({} as any),
+      });
+      jest
+        .spyOn(
+          indicatorRecordRepository,
+          'getIndicatorRawDataByGeoRegionAndMaterial',
+        )
+        .mockResolvedValue({
+          production: 10,
+          harvestedArea: 100,
+          rawBiodiversity: 10,
+          rawCarbon: 5,
+          rawDeforestation: 20,
+          rawWater: 25,
+        });
+
+      //ACT
+      await indicatorRecordService.createIndicatorRecordsBySourcingRecords(
+        sourcingData,
+      );
+
+      //ASSERT
+      const allIndicators = await indicatorRecordRepository.find();
+      expect(allIndicators.length).toEqual(4);
+
+      await checkCreatedIndicatorRecord(
+        INDICATOR_TYPES.DEFORESTATION,
+        indicatorPreconditions.deforestation,
+        sourcingData,
+        1000,
+        10,
+      );
+      await checkCreatedIndicatorRecord(
+        INDICATOR_TYPES.BIODIVERSITY_LOSS,
+        indicatorPreconditions.biodiversityLoss,
+        sourcingData,
+        500,
+        10,
+      );
+      await checkCreatedIndicatorRecord(
+        INDICATOR_TYPES.CARBON_EMISSIONS,
+        indicatorPreconditions.carbonEmissions,
+        sourcingData,
+        250,
+        10,
+      );
+      await checkCreatedIndicatorRecord(
+        INDICATOR_TYPES.UNSUSTAINABLE_WATER_USE,
+        indicatorPreconditions.waterRisk,
+        sourcingData,
+        12500,
+        10,
+      );
+    });
+  });
+
+  async function checkCreatedIndicatorRecord(
+    indicatorType: INDICATOR_TYPES,
+    h3Data: H3Data,
+    sourcingData: any,
+    recordValue: number,
+    scalerValue: number | null,
+  ): Promise<void> {
+    const indicatorRecords = await indicatorRecordRepository.find({
+      where: { indicatorId: h3Data.indicatorId },
+    });
+    expect(indicatorRecords.length).toEqual(1);
+    expect(indicatorRecords[0].sourcingRecordId).toEqual(
+      sourcingData.sourcingRecordId,
+    );
+    expect(indicatorRecords[0].status).toEqual(INDICATOR_RECORD_STATUS.SUCCESS);
+    expect(indicatorRecords[0].value).toEqual(recordValue);
+    expect(indicatorRecords[0].scaler).toEqual(scalerValue);
+    expect(indicatorRecords[0].h3DataId).toEqual(h3Data.id);
+    //Inidicator Coefficientes are not checked because it's not used
+  }
+
+  async function deleteH3GridTables(): Promise<void> {
+    const tableList: any[] = await getManager().query(`select table_name
```

Comment:
I just loved this. It will tide us to use some fixed table names for preconditions, but since the createWorldForIndicator.... mock seed function is there and we are using it, I think it is safe for now leaving it

Just a note, should we extract these functionality to a external module? So we can use it in more suites

Reply:
+1, we could use the same approach to test Indicator Records created by Scenario Intervention